### ISSUE TITLE
Window Controls not shown on second monitor ('Show taskbar on all displays' is Off)

### ISSUE THREAD

**Issue by schollexpos** [musescore/MuseScore#19942]
### Issue type

UI bug

### Bug description

The window controls (minimize, maximize and close) are not shown in the upper right corner of the musescore window (and can't be interacted with) when the window is launched on or moved to second monitor in maximized view.

### Steps to reproduce

1. Open Musescore
2. Move Window to second monitor
3. Maximize window
4. The Window Controls are not displayed anymore

### Screenshots/Screen recordings

![grafik](https://github.com/musescore/MuseScore/assets/25752700/fb027a9f-a28d-45e2-b349-471ae5b6ebf9)

(scores removed from screenshot by me)

### MuseScore Version

OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore version (64-bit): 4.1.1-232071203, revision: github-musescore-musescore-e4d1ddf

### Regression

No.

### Operating system

Windows 10

### Additional context

My System: Windows 11 with two monitors with 1920x1080 and same DPI

**Comment by cbjeukendrup** [comment]
Related to #19809

**Comment by iainhallam** [comment]
@schollexpos does pressing F11 to exit full-screen mode get the controls back for you?

**Comment by zacjansheski** [comment]
@schollexpos Is this issue still actual in 4.3.2 / master?

**Comment by iainhallam** [comment]
@zacjansheski 4.3.2 still does this on Windows 10.

**Comment by cbjeukendrup** [comment]
And the `master` nightly builds from https://musescore.org/en/nightly-builds?

**Comment by iainhallam** [comment]
Yup - OS: Windows 10 Version 2009 or later, Arch.: x86_64, MuseScore Studio version (64-bit): 4.4.0-241850505, revision: github-musescore-musescore-d46107f has the same behaviour. It opens on a secondary screen full-screen with no controls, and requires F11 to come out of that mode. Dragging it to the top of the main screen, it maximises, with controls visible; dragging to the top of a secondary screen, it again goes full-screen.

**Comment by davidlbangs** [comment]
I wanted to experiment whether it mattered which monitor was the "primary" model.  In MuseScore 4.4 and Windows 11 24H2, if I make the vertical monitor my "Primary" display, MuseScore can maximize correctly on that display, but not no longer maximize correctly on my laptop's built in monitor, because it is no longer the primary display.  

The scale of the two monitors is the same at 150% (recommended), so it is not a scale different.

**Comment by davidlbangs** [comment]
MuseScore 4.4.1 does not handle the act of changing the primary display well at all.  If I change my primary display to a different monitor then  launch MuseScore again, it does not draw its window at all.  I can not find its window until I set the primary monitor back to the same it used to be.

Now, after changing primary display, MuseScore is not operable but has a kind of ghostly image on my screen showing a distorted view of what it would look like it, but it is just a screen garbage image instead of a real app. image pasted.

<img width="1078" alt="Screenshot 2024-09-08 214342" src="https://github.com/user-attachments/assets/6ffd255a-b43c-470f-a4b3-9d2c1c5c8cff">

In this case, I just have to force quit MuseScore and relaunch it and it does appear normally.

**Comment by davidlbangs** [comment]
It occurs to me that MuseScore is remembering its screen position relative to whatever is the primary monitor, instead of remembering it relative to the window canvas like every other app does.  So, if I change my primary monitor to the right monitor while MuseScore is on the right monitor, the window position is interpreted as being off to the right of my right monitor.  If it is on my left monitor when I change the primary monitor from right to left, it is now off to the left of the left monitor.  

This sounds off topic to this forum, but I think it is all related.  MuseScore is paying attention to which monitor is primary, and it really should not.  All the monitors are arranged relatively to a canvas, not to the primary monitor.

**Comment by jernanigans** [comment]
Can this get prioritized higher? I commonly have MuseScore maximized on a secondary display, and sometimes I want to move it back to the primary display. The entire title bar becomes unclickable after MuseScore 4 is maximized on a secondary display. The workarounds, which are not shared here, are...

- Right click on MS4 instance in taskbar and click "Restore" to bring it back to windowed mode
- Using [Windows Key] + Left Arrow / Right Arrow to move the window back to the primary monitor.

I would love to just be able to drag the window back from secondary to primary. Lack of support for things like this are why I still use MuseScore 3.6.2 more often than MuseScore 4.

**Comment by Eism** [comment]
@jernanigans @davidlbangs @iainhallam could you please test the build from https://github.com/musescore/MuseScore/pull/28517?
Direct link to Windows build: https://github.com/musescore/MuseScore/actions/runs/15776851409/artifacts/3369337423

**Comment by DmitryArefiev** [comment]
Fixed in #28517
